app/server/lib/OIDCConfig.ts

```diff
@@ -155,14 +175,22 @@ export class OIDCConfig {
     return codeVerifier;
   }
 
-  private _makeUserProfileFromUserInfo(userInfo: UserinfoResponse) {
-      const email = userInfo.email;
-      const fname = userInfo.given_name ?? '';
-      const lname = userInfo.family_name ?? '';
-      return {
-        email,
-        name: `${fname} ${lname}`.trim(),
-      };
+  private _makeUserProfileFromUserInfo(userInfo: UserinfoResponse): Partial<UserProfile> {
+    return {
+      email: userInfo[ this._emailPropertyKey! ] as string,
```

Comment:
I suggest `String(userInfo[this._emailPropertyKey])` to ensure it's actually a string.